lib/vanagon/project.rb

```diff
@@ -692,5 +718,36 @@ def load_upstream_settings(upstream_project_name, upstream_git_url, upstream_git
         upstream_project.cleanup
       end
     end
+
+    # Load the settings hash for the current project/platform combination from a
+    # yaml file as produced by `publish_yaml_settings`. file:// and http:// URIs
+    # are accepted. If the URI uses http://, a sha1 URI is also required.
+    #
+    # @param settings_uri [String] A URI to a yaml settings file
+    # @param settings_sha1_uri [String] A URI to a sha1sum file for the yaml settings file
+    # @raise [Vanagon::Error] when the settings file can't be found
+    def load_yaml_settings(settings_uri, settings_sha1_uri = nil) # rubocop:disable Metrics/AbcSize
```

Comment:
This method was getting pretty long -- I've disabled the `AbcSize` metric here, since nothing jumped out at me as an obvious thing to extract into its own method, but I'm open to suggestions.